tests/host_reference/CloverForce_reference.h

```diff
@@ -0,0 +1,985 @@
+#pragma once
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "gauge_force_reference.h"
+#include "misc.h"
+#include <color_spinor_field.h> // convenient quark field container
+#include <command_line_params.h>
+#include <dslash_reference.h>
+#include <gauge_field.h>
+#include <host_utils.h>
+#include <quda.h>
+// #include <quda_matrix.h>
+// #include <complex_quda.h>
+#include <Eigen/Dense>
+
+// FIXME: this was copied from  wilson_dslash_reference.cpp maybe it is better to create a separate file with the projection
+// clang-format off
+static const double projector[8][4][4][2] = {
+  {
+    {{1,0}, {0,0}, {0,0}, {0,-1}},
+    {{0,0}, {1,0}, {0,-1}, {0,0}},
+    {{0,0}, {0,1}, {1,0}, {0,0}},
+    {{0,1}, {0,0}, {0,0}, {1,0}}
+  },
+  {
+    {{1,0}, {0,0}, {0,0}, {0,1}},
+    {{0,0}, {1,0}, {0,1}, {0,0}},
+    {{0,0}, {0,-1}, {1,0}, {0,0}},
+    {{0,-1}, {0,0}, {0,0}, {1,0}}
+  },
+  {
+    {{1,0}, {0,0}, {0,0}, {1,0}},
+    {{0,0}, {1,0}, {-1,0}, {0,0}},
+    {{0,0}, {-1,0}, {1,0}, {0,0}},
+    {{1,0}, {0,0}, {0,0}, {1,0}}
+  },
+  {
+    {{1,0}, {0,0}, {0,0}, {-1,0}},
+    {{0,0}, {1,0}, {1,0}, {0,0}},
+    {{0,0}, {1,0}, {1,0}, {0,0}},
+    {{-1,0}, {0,0}, {0,0}, {1,0}}
+  },
+  {
+    {{1,0}, {0,0}, {0,-1}, {0,0}},
+    {{0,0}, {1,0}, {0,0}, {0,1}},
+    {{0,1}, {0,0}, {1,0}, {0,0}},
+    {{0,0}, {0,-1}, {0,0}, {1,0}}
+  },
+  {
+    {{1,0}, {0,0}, {0,1}, {0,0}},
+    {{0,0}, {1,0}, {0,0}, {0,-1}},
+    {{0,-1}, {0,0}, {1,0}, {0,0}},
+    {{0,0}, {0,1}, {0,0}, {1,0}}
+  },
+  {
+    {{1,0}, {0,0}, {-1,0}, {0,0}},
+    {{0,0}, {1,0}, {0,0}, {-1,0}},
+    {{-1,0}, {0,0}, {1,0}, {0,0}},
+    {{0,0}, {-1,0}, {0,0}, {1,0}}
+  },
+  {
+    {{1,0}, {0,0}, {1,0}, {0,0}},
+    {{0,0}, {1,0}, {0,0}, {1,0}},
+    {{1,0}, {0,0}, {1,0}, {0,0}},
+    {{0,0}, {1,0}, {0,0}, {1,0}}
+  }
+};
+// clang-format on
+
+static const double local_gamma[4][4][4][2] = {{// x
```

Comment:
Missing `// clang-format off` and/or the `// clang-format on` above can be removed

Reply:
4954aff24fdf497bf043b86ccdbd3bafeb3d319c